### PR TITLE
Feat/private page

### PR DESCRIPTION
**Pull request by MacenaAndre**
- PrivatePage.js created
- localstorage.js created
- redirecting logged user to route "/timeline"
- redirecting unlogged user to route "/"

### CODE REVIEW

File: src/components/commons/PrivatePage.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        return (
+            <></>
+        );
+    }
+ 
+    return (
+        <>
+            {children};
+        </>
+    );
```
Comment (Francksuel) [inline src/components/commons/PrivatePage.js:35]: Great!

**Review by Francksuel — APPROVED** [review]
Well done!

File: src/components/commons/PrivatePage.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const {config, setConfig, userData, setUserData} = useContext(LoginContext);
+
+    const data = JSON.parse(localStorage.getItem("linkr"));
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if(!data) {
+            if(window.confirm("Please Login to continue")) {
+                navigate("/")
```
Comment (SauloDriven) [inline src/components/commons/PrivatePage.js:16]: **Suggestion:** it makes more sense that this file is inside a folder called decorator

File: src/services/localstorage.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function getUser() {
+    const data = localStorage.getItem("linkr");
+    return data ? JSON.parse(data) : null;
+};
+
+function setUser(data) {
+    return localStorage.setItem("linkr", JSON.stringify(data));
+};
+
+export {getUser, setUser};
```
Comment (SauloDriven) [inline src/services/localstorage.js:10]: **Suggestion:** it makes more sense that this file is inside a folder called utils

File: src/components/App.js [hunk 4]
```diff
@@ -35,9 +29,21 @@ export default function App() {
...
           <Route path="/" element={<SignIn />} />
           <Route path="/sign-up" element={<SignUp />} />
-          <Route path="/timeline" element={<Timeline />} />
-          <Route path="/hashtag/:hashtag" element={<Hashtag />} />
-          <Route path="/user/:id" element={<UserTimeline />} />
+          <Route path="/timeline" element={
+            <PrivatePage>
+                <Timeline />
+            </PrivatePage>
+          } />
```
Comment (SauloDriven) [inline src/components/App.js:36]: **Suggestion:** I believe that the code in the routes part is cleaner if you encompass this page before setting the element
```
const PrivateTimeLine =  <PrivatePage> <Timeline /> </PrivatePage>
```
```suggestion
           <PrivateTimeLine/> } />
```

**Review by SauloDriven — APPROVED** [review]
I gave some suggestions, if you have questions, contact me.

**Comment by MacenaAndre** [comment]
Noted the suggestions and will refactor as soon as i finish coding the deletePost requisition in the backend

**Pull request merged**